domain/status/state/state.go

```diff
@@ -40,6 +41,39 @@ func NewState(factory database.TxnRunnerFactory, clock clock.Clock, logger logge
 	}
 }
 
+// GetModelInfo returns only basic model information used for displaying model status.
+// The following error types can be expected to be returned:
+// - [modelerrors.NotFound]: When the model does not exist.
+func (st *State) GetModelInfo(ctx context.Context) (status.ModelStatusInfo, error) {
+	db, err := st.DB()
+	if err != nil {
+		return status.ModelStatusInfo{}, errors.Capture(err)
+	}
+
+	stmt, err := st.Prepare(`
+SELECT &modelInfo.*
+FROM   model
+`, modelInfo{})
+	if err != nil {
+		return status.ModelStatusInfo{}, errors.Capture(err)
+	}
+
+	var m modelInfo
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		if err == sqlair.ErrNoRows {
```

Comment:
Which error are we checking here?
Should we check the error returned from the below query?

Reply:
And we should use errors.Is.

Reply:
Missed this, thanks for pointing it out kelvin!